src/elements/content-explorer/ContentExplorer.js

```diff
@@ -1330,14 +1359,28 @@ class ContentExplorer extends Component<Props, State> {
     };
 
     /**
-     * Handle pagination changes
+     * Handle pagination changes for offset based pagination
      *
      * @param {number} newOffset - the new page offset value
      */
     paginate = (newOffset: number) => {
         this.setState({ currentOffset: newOffset }, this.refreshCollection);
     };
 
+    /**
+     * Handle pagination changes for marker based pagination
+     * @param {number} newOffset - the new page offset value
+     */
+    markerBasedPaginate = (newOffset: number) => {
+        const { currentPageNumber } = this.state;
+        this.setState(
+            {
+                currentPageNumber: currentPageNumber + newOffset, // newOffset could be negative
+            },
+            this.refreshCollection,
+        );
```

Comment:
`currentPageNumber` is used as an index to get the marker from the `[]` to fetch next set of results.